From 8259734c58bdcf2ebb06f830b97af09a659ea681 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 16 Jun 2024 07:38:14 +0200
Subject: [PATCH] Linux patch

---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index e3cedca..d95d8f4 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ include Makethird
 # invocation line - specify XCFLAGS, XLDFLAGS, XLIB_LDFLAGS, XEXE_LDFLAGS or
 # XLIBS instead. Make ignores any lines in the makefile that set a variable
 # that was set on the command line.
-CFLAGS += $(XCFLAGS) -Iinclude
+CFLAGS += $(XCFLAGS) -Iinclude -fPIC
 LIBS += $(XLIBS) -lm
 
 LDFLAGS += $(XLDFLAGS)
@@ -492,10 +492,10 @@ install-docs:
 	install -m 644 docs/man/*.1 $(DESTDIR)$(mandir)/man1
 
 	install -d $(DESTDIR)$(docdir)
-	install -d $(DESTDIR)$(docdir)/examples
+	# install -d $(DESTDIR)$(docdir)/examples
 	install -m 644 README CHANGES $(DESTDIR)$(docdir)
-	install -m 644 $(wildcard COPYING LICENSE) $(DESTDIR)$(docdir)
-	install -m 644 docs/examples/* $(DESTDIR)$(docdir)/examples
+	# install -m 644 $(wildcard COPYING LICENSE) $(DESTDIR)$(docdir)
+	# install -m 644 docs/examples/* $(DESTDIR)$(docdir)/examples
 
 install: install-libs install-apps install-docs
 
