From 4d4a5423abfc0965b21482aa90c332a7a231f251 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Sun, 16 Jun 2024 07:45:00 +0200
Subject: [PATCH] Update Makefile

---
 Makefile | 55 +++++++++++++++----------------------------------------
 1 file changed, 15 insertions(+), 40 deletions(-)

diff --git a/Makefile b/Makefile
index e3cedca..a33a47a 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ include Makethird
 # invocation line - specify XCFLAGS, XLDFLAGS, XLIB_LDFLAGS, XEXE_LDFLAGS or
 # XLIBS instead. Make ignores any lines in the makefile that set a variable
 # that was set on the command line.
-CFLAGS += $(XCFLAGS) -Iinclude
+CFLAGS += $(XCFLAGS) -Iinclude -fPIC
 LIBS += $(XLIBS) -lm
 
 LDFLAGS += $(XLDFLAGS)
@@ -298,47 +298,28 @@ generate: source/html/css-properties.h
 
 # --- Library ---
 
-ifeq ($(shared),yes)
-MUPDF_LIB = $(OUT)/libmupdf.$(SO)$(SO_VERSION)
-ifeq ($(SO),dll)
-MUPDF_LIB_IMPORT = $(OUT)/libmupdf_$(SO).a
-LIBS_TO_INSTALL_IN_BIN = $(MUPDF_LIB)
-LIBS_TO_INSTALL_IN_LIB = $(MUPDF_LIB_IMPORT)
-else
+MUPDF_LIB = $(OUT)/libmupdf.dylib
 LIBS_TO_INSTALL_IN_LIB = $(MUPDF_LIB)
-endif
-ifneq ($(USE_SYSTEM_GLUT),yes)
-THIRD_GLUT_LIB = $(OUT)/libmupdf-glut.a
-endif
-THREAD_LIB = $(OUT)/libmupdf-threads.a
-PKCS7_LIB = $(OUT)/libmupdf-pkcs7.a
-
-$(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_OBJ)
-$(THIRD_GLUT_LIB) : $(THIRD_GLUT_OBJ)
-$(THREAD_LIB) : $(THREAD_OBJ)
-$(PKCS7_LIB) : $(PKCS7_OBJ)
-else
-MUPDF_LIB = $(OUT)/libmupdf.a
-LIBS_TO_INSTALL_IN_LIB = $(MUPDF_LIB) $(THIRD_LIB)
-THIRD_LIB = $(OUT)/libmupdf-third.a
-ifneq ($(USE_SYSTEM_GLUT),yes)
-THIRD_GLUT_LIB = $(OUT)/libmupdf-glut.a
-endif
-THREAD_LIB = $(OUT)/libmupdf-threads.a
-PKCS7_LIB = $(OUT)/libmupdf-pkcs7.a
+THREAD_LIB = $(OUT)/libmupdf-threads.dylib
+PKCS7_LIB = $(OUT)/libmupdf-pkcs7.dylib
+THIRD_LIB = $(OUT)/libmupdf-third.dylib
 
 $(MUPDF_LIB) : $(MUPDF_OBJ)
 $(THIRD_LIB) : $(THIRD_OBJ)
-$(THIRD_GLUT_LIB) : $(THIRD_GLUT_OBJ)
 $(THREAD_LIB) : $(THREAD_OBJ)
 $(PKCS7_LIB) : $(PKCS7_OBJ)
-endif
 
-$(MUPDF_LIB) : $(MUPDF_OBJ)
+$(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_LIB) $(THREAD_LIB)
+	$(LINK_CMD) -shared -Wl,-install_name -Wl,libmupdf.dylib -undefined dynamic_lookup $(THIRD_LIBS)
 $(THIRD_LIB) : $(THIRD_OBJ)
+	$(LINK_CMD) -shared -Wl,-install_name -Wl,libmupdf-third.dylib -undefined dynamic_lookup
 $(THREAD_LIB) : $(THREAD_OBJ)
+	$(LINK_CMD) -shared -Wl,-install_name -Wl,libmupdf-threads.dylib -undefined dynamic_lookup -lpthread
 $(PKCS7_LIB) : $(PKCS7_OBJ)
+	$(LINK_CMD) -v -shared -Wl,-install_name -Wl,libmupdf-pkcs7.dylib -undefined dynamic_lookup
 
+LIBS_TO_INSTALL_IN_LIB := $(MUPDF_LIB) $(THIRD_LIB) $(THREAD_LIB) $(PKCS7_LIB) 
+ 
 # --- Main tools and viewers ---
 
 MUTOOL_SRC := source/tools/mutool.c
@@ -492,10 +473,10 @@ install-docs:
 	install -m 644 docs/man/*.1 $(DESTDIR)$(mandir)/man1
 
 	install -d $(DESTDIR)$(docdir)
-	install -d $(DESTDIR)$(docdir)/examples
+	# install -d $(DESTDIR)$(docdir)/examples
 	install -m 644 README CHANGES $(DESTDIR)$(docdir)
-	install -m 644 $(wildcard COPYING LICENSE) $(DESTDIR)$(docdir)
-	install -m 644 docs/examples/* $(DESTDIR)$(docdir)/examples
+	# install -m 644 $(wildcard COPYING LICENSE) $(DESTDIR)$(docdir)
+	# install -m 644 docs/examples/* $(DESTDIR)$(docdir)/examples
 
 install: install-libs install-apps install-docs
 
@@ -645,16 +626,10 @@ endif
 install-shared-c: install-shared-check shared install-headers
 	install -d $(DESTDIR)$(libdir)
 	install -m $(SO_INSTALL_MODE) $(OUT)/libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/
-ifneq ($(OS),OpenBSD)
-	ln -sf libmupdf.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdf.$(SO)
-endif
 
 install-shared-c++: install-shared-c c++
 	install -m 644 platform/c++/include/mupdf/*.h $(DESTDIR)$(incdir)/mupdf
 	install -m $(SO_INSTALL_MODE) $(OUT)/libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/
-ifneq ($(OS),OpenBSD)
-	ln -sf libmupdfcpp.$(SO)$(SO_VERSION) $(DESTDIR)$(libdir)/libmupdfcpp.$(SO)
-endif
 
 install-shared-python: install-shared-c++ python
 	install -d $(DESTDIR)$(pydir)/mupdf
